### PR TITLE
Try fallback auto-merge when unknown protection

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
This edge case happens when the access token used doesn't have enough permissions to check whether branch protection is enabled or not. See [6337623700](https://github.com/iterative/discord-corpus/runs/6337623700?check_suite_focus=true) for an example.

### CODE REVIEW

File: src/drivers/github.js [hunk 1]
```diff
@@ -447,13 +447,20 @@ class Github {
...
+          winston.warn(
+            `Failed to enable auto-merge: Enable the feature in your repository settings: ${settingsUrl}#merge_types_auto_merge. Trying to merge immediately...`
+          );
+        } else {
+          winston.warn(
+            `Failed to enable auto-merge: Set up branch protection and add "required status checks" for branch '${base}': ${settingsUrl}/branches. Trying to merge immediately...`
+          );
+        }
+      } catch (err) {
+        if (!err.message.includes('Resource not accessible by integration')) throw err;
```
Comment (0x2b3bfa0) [inline src/drivers/github.js:461]:
```suggestion
        if (!err.message.includes('Resource not accessible by integration'))
          throw err;
```

**Comment by DavidGOrtega** [comment]
@0x2b3bfa0 is this still going on?

**Review by casperdcl — APPROVED** [review]
lgtm but it seems like a GH quirk that may change in future? How would we monitor this?

**Comment by 0x2b3bfa0** [comment]
> [...] _it seems like a GH quirk that may change in future? How would we monitor this?_

Updated https://github.com/iterative/cml/pull/968#issue-1206260461

**Pull request merged**